waku/v2/protocol/rln/group_manager/dynamic/web3.go

```diff
@@ -34,7 +34,7 @@ func (gm *DynamicGroupManager) HandleGroupUpdates(ctx context.Context, handler R
 			return errors.New("persisted data: contract address mismatch")
 		}
 
-		fromBlock = metadata.LastProcessedBlock
+		fromBlock = metadata.LastProcessedBlock + 1
```

Comment:
Yes, it seems okay to me!
Interestingly enough, with the code as it was before, loading the same block twice even if that block has a membership event, did not cause problems because the RLN merkle tree wasn't affected because it will reinsert the same data in the same leaf index.